project-management/src/main/java/life/qbic/projectmanagement/domain/model/measurement/NGSMeasurement.java

```diff
@@ -114,35 +114,58 @@ private static void evaluateMandatoryMetadata(NGSMethodMetadata method)
+   * @return an instance of an {@link NGSMeasurement}
    * @since 1.0.0
    */
-  public static NGSMeasurement create(ProjectId projectId, Collection<SampleId> sampleIds,
+  public static NGSMeasurement createWithPool(ProjectId projectId, String samplePool,
       MeasurementCode measurementCode, Organisation organisation, NGSMethodMetadata method,
-      String comment) throws IllegalArgumentException {
-    if (sampleIds.isEmpty()) {
-      throw new IllegalArgumentException(
-          "No sample ids provided. At least one sample id must provided for a measurement.");
+      Collection<NGSSpecificMeasurementMetadata> specificMeasurementMetadata)
+      throws IllegalArgumentException {
+    requireNonNull(measurementCode);
+    requireNonNull(method);
+    requireNonNull(method.instrument());
+    if (samplePool.isBlank()) {
+      throw new IllegalArgumentException("Sample Pool: no value provided");
     }
-    requireNonNull(measurementCode, "measurement code must not be null");
-    requireNonNull(method, "method must not be null");
-    requireNonNull(method.instrument(), "instrument must not be null");
-    if (!measurementCode.isNGSDomain()) {
+    if (specificMeasurementMetadata.stream().map(NGSSpecificMeasurementMetadata::index)
+        .anyMatch(Optional::isEmpty)) {
       throw new IllegalArgumentException(
-          "NGSMeasurementMetadata code is not from the NGS domain for: \"" + measurementCode
-              + "\"");
+          "All specific metadata must have an index in a pooled measurement");
     }
     var measurementId = MeasurementId.create();
-    return new NGSMeasurement(measurementId, projectId, sampleIds, measurementCode, organisation,
-        method,
-        comment, Instant.now());
+    return new NGSMeasurement(measurementId, projectId, samplePool, measurementCode, organisation,
+        method, Instant.now(), specificMeasurementMetadata);
+  }
+
+  /**
+   * Creates a new single {@link NGSMeasurement} object instance, that describes an NGS measurement
+   * entity with many describing properties about provenance and instrumentation.
+   *
+   * @param projectId                   the project id the measurement belongs to
+   * @param measurementCode             the assigned measurement code for the measurement
+   * @param organisation                where the measurement has been conducted
+   * @param method                      measurement method related metadata
+   * @param specificMeasurementMetadata sample specific metadata
+   * @return an instance of an {@link NGSMeasurement}
+   * @since 1.0.0
+   */
+  public static NGSMeasurement createSingleMeasurement(ProjectId projectId,
+      MeasurementCode measurementCode, Organisation organisation, NGSMethodMetadata method,
+      NGSSpecificMeasurementMetadata specificMeasurementMetadata) {
+    requireNonNull(measurementCode);
+    requireNonNull(method);
+    requireNonNull(method.instrument());
```

Comment:
NIT: use the error message in the `Objects#requireNonNull`

Reply:
Nice catch 👀